src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -128,8 +127,8 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
 
         private void LogSuccessfulTelemetry(AuthenticationResult authenticationResult, MsalTelemetryEventDetails telemetryEventDetails)
         {
-            telemetryEventDetails.SetProperty(TelemetryConstants.CacheInfoTelemetry, authenticationResult.AuthenticationResultMetadata.CacheRefreshReason.ToString());
-            telemetryEventDetails.SetProperty(TelemetryConstants.TokenSource, authenticationResult.AuthenticationResultMetadata.TokenSource.ToString());
+            telemetryEventDetails.SetProperty(TelemetryConstants.CacheInfoTelemetry, Convert.ToInt64(authenticationResult.AuthenticationResultMetadata.CacheRefreshReason));
```

Comment:
why int64 (i.e. long) and not int32?

Reply:
The method to set property does not have an iverload for int32. Only for long. Looks like a limitation of Aria SDK. @TimHannMSFT can confirm this.

Reply:
Yes, the AriaSDK only supports long natively which we decided to reflect in the basic contract.